core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -470,24 +507,56 @@ public static Timer timer(String name) {
    * The synchronized keyword is added for correctness with {@link #resetAllMetrics}
    */
   private static synchronized List<alluxio.grpc.Metric> reportMetrics(InstanceType instanceType) {
+    if (!sReported) {
+      initShouldReportMetrics(instanceType);
+      sReported = true;
+    }
     List<alluxio.grpc.Metric> rpcMetrics = new ArrayList<>(20);
-    for (Metric m : allInstanceMetrics(instanceType)) {
```

Comment:
I want to remove the logics of calling original `allMetrics` and `allInstanceMetrics`. These two methods do many full metrics traverse and do many metric copies which takes up a lot of the heap size.

`METRIC_REGISTERY.getGauges(), getCounters(), getMeters(), getTimers()` will internally traverse the `MetricRegistry.metrics` and make a copy of metrics belonging to the specific metric type.
Change to use  one `METRIC_REGISTRY.getMetrics()` will directly return the UnmodifiableSortedMap view of the original `MetricRegistry.metrics` map  which reduces four traverse of the `MetricRegistry.metrics` map  and reduce a full copy of the full metrics map.

Reply:
Original report Metrics:  four traverse of full metrics, one copy of the full metrics, one copy of the worker or client metrics, one copy of the to be reported metrics.
Now report metrics: one traverse of the should report metrics, one copy of the to be reported metrics.